Formula/glibc.rb

```diff
@@ -95,43 +95,166 @@ class Glibc < Formula
+    sha256 "c9297d5bcd7cb43f3dfc2fed5389e948c9312fd962ef6a4ce455cff963ebe4f1"
+  end
+
+  resource "mpfr" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpfr-3.1.6.tar.bz2"
+    sha256 "cf4f4b2d80abb79e820e78c8077b6725bbbb4e8f41896783c899087be0e94068"
+  end
+
+  resource "mpc" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpc-1.0.3.tar.gz"
+    sha256 "617decc6ea09889fb08ede330917a00b16809b8db88c29c31bfbb49cbf88ecc3"
+  end
+
+  resource "gmp" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/gmp-6.1.0.tar.bz2"
+    sha256 "498449a994efeba527885c10405993427995d3f86b8768d8cdf8d9dd7c6b73e8"
+  end
+
+  resource "make" do
+    url "https://ftp.gnu.org/gnu/make/make-4.3.tar.gz"
+    sha256 "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
+  end
 
+  resource "python3" do
+    url "https://www.python.org/ftp/python/3.10.5/Python-3.10.5.tgz"
+    sha256 "18f57182a2de3b0be76dfc39fdcfd28156bb6dd23e5f08696f7492e9e3d0bf2d"
+  end
+
+  resource "m4" do
+    url "https://ftp.gnu.org/gnu/m4/m4-1.4.19.tar.xz"
+    sha256 "63aede5c6d33b6d9b13511cd0be2cac046f2e70fd0a07aa9573a04a82783af96"
+  end
+
+  resource "bison" do
+    url "https://ftp.gnu.org/gnu/bison/bison-3.8.2.tar.xz"
+    sha256 "9bba0214ccf7f1079c5d59210045227bcf619519840ebfa80cd3849cff5a5bf2"
+  end
+
+  def install
     # Setting RPATH breaks glibc.
     %w[
       LDFLAGS LD_LIBRARY_PATH LD_RUN_PATH LIBRARY_PATH
       HOMEBREW_DYNAMIC_LINKER HOMEBREW_LIBRARY_PATHS HOMEBREW_RPATH_PATHS
     ].each { |x| ENV.delete x }
 
+    # TODO: make an audit exception or come up with a better way
+    # to use /usr/bin/gcc.
+    mkdir buildpath/"gcc/bin"
+    ln_sf "/usr/bin/gcc", buildpath/"gcc/bin"
+    ln_sf "/usr/bin/g++", buildpath/"gcc/bin"
+    ENV["CC"] = buildpath/"gcc/bin/gcc"
+    ENV["CXX"] = buildpath/"gcc/bin/g++"
+    ENV.prepend_path "PATH", buildpath/"bootstrap/bin"
+
+    # TODO: only run if make is missing or too old
+    resource("make").stage do
+      system "./configure", "--prefix=#{buildpath}/bootstrap"
+      system "make", "install"
+    end
+    ENV["MAKE"] = buildpath/"bootstrap/bin/make"
+
+    if which("python3").blank?
+      resource("python3").stage do
+        with_env(
+          CPPFLAGS: "-I#{Formula["zlib"].opt_include}",
+          LDFLAGS:  "-L#{Formula["zlib"].opt_lib} -Wl,-rpath,#{Formula["zlib"].opt_lib}",
+        ) do
+          system "./configure", "--prefix=#{buildpath}/bootstrap"
+          system "make"
+          system "make", "install"
+        end
+      end
+    end
+
+    if which("bison").blank?
+      if which("m4").blank?
+        resource("m4").stage do
+          system "./configure", "--prefix=#{buildpath}/bootstrap"
+          system "make"
+          system "make", "install"
+        end
+      end
+
+      resource("bison").stage do
+        system "./configure", "--prefix=#{buildpath}/bootstrap",
+                              "M4=#{buildpath}/bootstrap/bin/m4"
```

Comment:
You probably only want to add this if you actually build `m4`. You might not have.

Reply:
Good call, I’ll add a conditional for this.